File: server/core/Field.js
var Field = class{
	constructor(){
		this.soldier = [];
		this.barrier = [];
	}

	setBarrier (card){
		this.barrier.push(card);
		this.barrier[this.barrier.length - 1].isCharge = true;
		this.barrier[this.barrier.length - 1].isFront = true;
	}
	
	summon (card){
		this.soldier.push(card);
		if(card.job == "ace" || card.job == "magician"){
			this.soldier[this.soldier.length - 1].canAttack = true;
		}
	}
	
	lookAttackable (){
		var list = [];
		for(var i = 0; i < this.soldier.length; i++){
			if(this.soldier[length].canAttack){
				list.push(this.soldier[length]);
			}
		}
		return list;
	}
	
	recoveryAll (){
		for(var i = 0; i < this.soldier.length; i++){
			this.recovery(i);
		}
	}
	
	recovery (length){
		this.soldier[length].canAttack = true;
	}
	
	chargeAll (){
		for(var i = 0; i < this.soldier.length; i++){
			this.charge("soldier", i);
		}
		
		for(i = 0; i < this.barrier.length; i++){
			this.charge("barrier", i);
		}
	}
	
	canAttackAll (){
		for(var i = 0; i < this.soldier.length; i++){
			this.soldier[i].canAttack = true;
		}
	}
	
	charge (place, length){
		if(place == "soldier"){
			this.soldier[length].isCharge = true;
		}else if(place == "barrier"){
			this.barrier[length].isCharge = true;
		}
	}
	
	drive (place, length){
		if(place == "soldier"){
			this.soldier[length].isCharge = false;
		}else if(place == "barrier"){
			this.barrier[length].isCharge = false;
		}
	}
	
	destruction (place, length){
		var card;
		if(place == "soldier"){
			card = this.soldier[length];
			this.soldier.splice(length, 1);
		}else if(place == "barrier"){
			card = this.barrier[length];
			this.barrier.splice(length, 1);
		}
		return card;
	}
	
	open (length){
		this.barrier[length].isFront = true;
	}
};

module.exports = Field;

/*
var Field = function(){
	this.soldier = [];
	this.barrier = [];
};

Field.prototype.setBarrier = function(card){
	this.barrier.push(card);
	this.barrier[this.barrier.length - 1].isCharge = true;
	this.barrier[this.barrier.length - 1].isFront = true;
};

Field.prototype.summon = function(card){
	this.soldier.push(card);
	if(card.job == "ace" || card.job == "magician"){
		this.soldier[this.soldier.length - 1].canAttack = true;
	}
};

Field.prototype.lookAttackable = function(){
	var list = [];
	for(var i = 0; i < this.soldier.length; i++){
		if(this.soldier[length].canAttack){
			list.push(this.soldier[length]);
		}
	}
	return list;
};

Field.prototype.recoveryAll = function(){
	for(var i = 0; i < this.soldier.length; i++){
		this.recovery(i);
	}
};

Field.prototype.recovery = function(length){
	this.soldier[length].canAttack = true;
};

Field.prototype.chargeAll = function(){
	for(var i = 0; i < this.soldier.length; i++){
		this.charge("soldier", i);
	}
	
	for(i = 0; i < this.barrier.length; i++){
		this.charge("barrier", i);
	}
};

Field.prototype.canAttackAll = function(){
	for(var i = 0; i < this.soldier.length; i++){
		this.soldier[i].canAttack = true;
	}
};

Field.prototype.charge = function(place, length){
	if(place == "soldier"){
		this.soldier[length].isCharge = true;
	}else if(place == "barrier"){
		this.barrier[length].isCharge = true;
	}
};

Field.prototype.drive = function(place, length){
	if(place == "soldier"){
		this.soldier[length].isCharge = false;
	}else if(place == "barrier"){
		this.barrier[length].isCharge = false;
	}
};

Field.prototype.destruction = function(place, length){
	var card;
	if(place == "soldier"){
		card = this.soldier[length];
		this.soldier.splice(length, 1);
	}else if(place == "barrier"){
		card = this.barrier[length];
		this.barrier.splice(length, 1);
	}
	return card;
};

Field.prototype.open = function(length){
	this.barrier[length].isFront = true;
};

module.exports = Field;
 */

File: Python_Server/bp/Field.py
import Card


class Field:
    def __init__(self):
        self.soldier = []
        self.barrier = []

    # 防壁をフィールドに追加
    def set_barrier(self, card: Card):
        self.barrier.append(card)
        self.barrier[len(self.barrier) - 1].is_charge = True
        self.barrier[len(self.barrier) - 1].is_front = True

    # 兵士をフィールドに追加
    def summon(self, card: Card):
        self.soldier.append(card)
        if card.job == "ace" or card.job == "magician":
            self.soldier[len(self.soldier) - 1].can_attack = True

    # 攻撃可能な兵士のリストを返す
    def look_attackable(self) ->list:
        list = []
        for i in range(0, len(self.soldier)):
            if self.soldier[i].can_attack:
                list.append(self.soldier[i])
        return list

    # 特定の兵士を攻撃可能にする
    def recovery(self, length: int):
        self.soldier[length].can_attack = True

    # 兵士全員を攻撃可能にする
    def recovery_all(self):
        for i in range(0, len(self.soldier)):
            self.recvery(i)

    # 兵士全員を攻撃可能にする
    def can_attack_all(self):
        for i in range(0, len(self.soldier)):
            self.soldier[i].can_attack = True

    # カードをチャージ状態にする
    def charge(self, place: str, length: int):
        if place == "soldier":
            self.soldier[length].is_charge = True
        elif place == "barrier":
            self.barrier[length].is_charge = True

    # カード全てをチャージ状態にする
    def charge_all(self):
        for i in range(0, len(self.soldier)):
            self.charge("soldier", i)

        for i in range(0, len(self.barrier)):
            self.charge("barrier", i)

    # カードをドライブ状態にする
    def drive(self, place: str, length: int):
        if place == "soldier":
            self.soldier[length].is_charge = False
        elif place == "barrier":
            self.barrier[length].is_charge = False

    # フィールドからカードを取り除く
    def destruction(self, place: str, length: int) ->Card:
        if place == "soldier":
            card = self.soldier[length]
            self.soldier.pop(length)
        elif place == "barrier":
            card = self.barrier[length]
            self.barrier.pop(length)

        return card

    # 防壁を表向きにする
    def open(self, length: int):
        self.barrier[length].is_front = True
